### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by Nadrieril**
This improves the logging setup to in particular include timing information for functions annotated with `#[tracing::instrument]`. To get these logs, call charon with `RUSTC_LOG=charon`.

### CODE REVIEW

File: charon/Cargo.toml [hunk 1]
```diff
@@ -69,7 +69,8 @@ stacker = "0.1"
 take_mut = "0.2.2"
 toml = { version = "0.8", features = ["parse"] }
 tracing-subscriber = { version = "0.3", features = [ "env-filter", "std", "fmt" ] }
-tracing-tree = "0.3"
+tracing-tree = { git = "https://github.com/Nadrieril/tracing-tree", features = [ "time" ] } # Fork with improved formating and timing info.
```
Comment (sonmarcho) [inline charon/Cargo.toml:72]: What is your plan for this fork?
If we keep this change, on the longer term I think we should make the fork live in the AeneasVerif repo.
What do you think?

File: charon/src/logger.rs [hunk 2]
```diff
@@ -11,130 +10,86 @@ pub fn initialize_logger() {
...
-            tracing_tree::HierarchicalLayer::new(2)
+            tracing_tree::HierarchicalLayer::new(1)
                 .with_ansi(true)
-                .with_indent_amount(1)
-                .with_indent_lines(true),
-        );
-    tracing::subscriber::set_global_default(subscriber).unwrap();
+                .with_indent_lines(true)
+                .with_bracketed_fields(true)
+                .with_timer(tracing_tree::time::Uptime::default()),
```
Comment (sonmarcho) [inline charon/src/logger.rs:22]: Do we print the local time of the machine, or the time elapsed since the start of the execution?

Comment (Nadrieril) [reply charon/src/logger.rs:22]: Each function call prints the time elapsed during that function call

File: charon/Cargo.toml (hunk 1 above)
Comment (Nadrieril) [reply charon/Cargo.toml:72]: The goal would be to upstream my changes to the main `tracing-tree` project. I'll experiment more in charon first though.

Importantly, note that this is a drop-in replacement for the upstream version. So we could revert to the upstream version and only use my fork locally when debugging, if using my fork like this is undesireable.

Comment (sonmarcho) [reply charon/Cargo.toml:72]: Upstreaming the changes to the main `tracing-tree` is probably the best thing to do.

**Pull request merged**